chore: add docs and e2e for alias (#2379)
aws/copilot-cli: site/content/docs/developing/domain.en.md

# Domain

As mentioned in [Application Guide](../concepts/applications.en.md#additional-app-configurations), you can configure your domain name of you app when doing `copilot app init`. Then after deploying your [Load Balanced Web Services](../concepts/services.en.md#load-balanced-web-service), you should be able to access your services publicly via

```
${SvcName}.${EnvName}.${AppName}.${DomainName}
```

For example:

```
https:kudo.test.coolapp.example.aws
```

> ## What is Copilot doing when using domain?

Reviewer: random thought: I feel like explanations like this (e.g. "what's happening under the hood?") should usually be placed at the bottom of the page.

> By default, under the hood Copilot

Reviewer: If this is the "default" behavior, do you think it worths explaining what would be the non-default options?

* creates a hosted zone in your app account for the new app subdomain `${AppName}.${DomainName}`;
* creates another hosted zone in your env account for the new env subdomain `${EnvName}.${AppName}.${DomainName}`;
* creates and validates an ACM certificate for the env subdomain;
* associates the certificate with your HTTPS listener and redirects HTTP traffic to HTTPS.

## How do I configure an alias for my service?

> If you don't like the default domain name Copilot assigns to your service, setting an alias for your service is also very easy. You can add it directly to your [manifest](../manifest/overview.en.md) `alias` section. The following snippet will set an alias to your service.

Reviewer: Could a brief explanation of what `alias` is be helpful here?

``` yaml
# in copilot/{service name}/manifest.yml
http:
  path: '/'
  alias: example.aws
```

> Note that for now you can only use aliases under the domain you specified when creating the application. We'll make this feature more powerful by allowing users to import certificates so as to use any alias in the future!

Reviewer: Do you think we need a `!!!info` for this?

Author: good idea!

# Domain

> As mentioned in [Application Guide](../concepts/applications.en.md#additional-app-configurations), you can configure your domain name of you app when doing `copilot app init`. Then after deploying your [Load Balanced Web Services](../concepts/services.en.md#load-balanced-web-service), you should be able to access your services publicly via

Reviewer 2: ```suggestion
As mentioned in the [Application Guide](../concepts/applications.en.md#additional-app-configurations), you can configure the domain name of your app when running `copilot app init`. After deploying your [Load Balanced Web Services](../concepts/services.en.md#load-balanced-web-service), you should be able to access them publicly via
```

```
${SvcName}.${EnvName}.${AppName}.${DomainName}
```

For example:

```
https:kudo.test.coolapp.example.aws
```

## How do I configure an alias for my service?

> If you don't like the default domain name Copilot assigns to your service, setting an [alias](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-choosing-alias-non-alias.html) for your service is also very easy. You can add it directly to your [manifest](../manifest/overview.en.md) `alias` section. The following snippet will set an alias to your service.

Reviewer 2: ```suggestion
If you don't like the default domain name Copilot assigns to your service, setting an [alias](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-choosing-alias-non-alias.html) for your service is also very easy. You can add it directly to your [manifest's](../manifest/overview.en.md) `alias` section. The following snippet will set an alias to your service.
```

``` yaml
# in copilot/{service name}/manifest.yml
http:
  path: '/'
  alias: example.aws
```

!!!info

>     Note that for now you can only use aliases under the domain you specified when creating the application. We'll make this feature more powerful by allowing users to import certificates so as to use any alias in the future!

Reviewer 2: ```suggestion
    Currently, you can only use aliases under the domain you specified when creating the application. We'll make this feature more powerful in the future by allowing you to import certificates and use any aliases!
```

## What happens under the hood?

> Under the hood Copilot

Reviewer 2: Nit: it may be too repetitive to have both the question and "Under the hood"...?
```suggestion
Under the hood, Copilot
```

> * creates a hosted zone in your app account for the new app subdomain `${AppName}.${DomainName}`;

Reviewer 2: Nit: because these are bullet points, I don't think we need the end punctuation-- the semi-colons and periods.

> * creates another hosted zone in your env account for the new env subdomain `${EnvName}.${AppName}.${DomainName}`;

Reviewer 2: I love that this section tells folks what is happening behind the scenes! And so much is happening!

  alias: example.aws
```

!!!info
    Note that for now you can only use aliases under the domain you specified when creating the application. We'll make this feature more powerful by allowing users to import certificates so as to use any alias in the future!

> ## What is Copilot doing when using domain?

Reviewer: Sorry I missed this in last round! Maybe something like "What happens under the hood?" like in https://aws.github.io/copilot-cli/docs/commands/storage-init/#what-happens-under-the-hood.
